<a href="https://colab.research.google.com/github/virajvaidya/SRC_ANU_Sampling_1/blob/main/HomeworkW5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [33]:
import pandas as pd
import numpy as np
import statistics as st
import itertools as it
from matplotlib import pyplot as plt
from scipy.stats import norm
from scipy.stats import t

In [8]:
populationSS = pd.read_excel('/content/drive/MyDrive/Sampling I - HW5 data.xls', sheet_name = 'Single-stage cluster sample')
populationTS = pd.read_excel('/content/drive/MyDrive/Sampling I - HW5 data.xls', sheet_name = 'Two-stage cluster sample')

## Q1 a Compute the BMI (BMI), its standard error and 95% confidence interval

In [9]:
populationSS

CLUSTER  Cluster total of BMI  Cluster total EVERPREG
0        1                  2875                      75
1        2                  2703                      73
2        3                  2701                      70
3        4                  2787                      70
4        5                  2743                      58
5        6                  2590                      74
6        7                  2647                      78
7        8                  2405                      57
8        9                  2762                      72
9       10                  2560                      56

In [12]:
populationSS.columns

Index(['CLUSTER', 'Cluster total of BMI', 'Cluster total EVERPREG'], dtype='object')

In [22]:
#PARAMETERS PROVIDED

n = 930    # Sample size
N = 9300   # Population size
A = 100    # Total number of clusters
B = 93     # Number of women per cluster

In [24]:
a = populationSS.shape[0]

In [27]:
mean_BMI = populationSS['Cluster total of BMI'].sum() / n
print(mean_BMI)

28.788172043010753


In [30]:
s2 = np.var(populationSS['Cluster total of BMI'], ddof=1)  # Sample variance of cluster totals
SE_mean_BMI = np.sqrt((1 / a) * s2) / B  # Standard error
print(SE_mean_BMI)

0.4537669797774879


In [35]:
t_crit = t.ppf(0.975, df=a - 1)  # t critical value for 95% CI
CI_lower = mean_BMI - t_crit * SE_mean_BMI
CI_upper = mean_BMI + t_crit * SE_mean_BMI

In [36]:
print(CI_lower)
print(CI_upper)

27.761679819440438
29.81466426658107


##Mean BMI : 28.7881
##Standard Error for CLustering: 0.4537
##Confidence Interval Lower: 27.761679
##Confidence Interval Upper: 29.814664


In [37]:
populationTS

CLUSTER  ELEMENT  BMI  EVERPREG
0         1        1   18         1
1         1        2   28         1
2         1        3   39         1
3         1        4   26         0
4         2        1   29         1
5         2        2   24         0
6         2        3   44         1
7         2        4   33         1
8         3        1   29         1
9         3        2   28         1
10        3        3   32         1
11        3        4   27         1
12        4        1   29         0
13        4        2   22         0
14        4        3   47         1
15        4        4   30         1
16        5        1   27         1
17        5        2   23         1
18        5        3   43         1
19        5        4   23         1
20        6        1   27         0
21        6        2   25         1
22        6        3   30         1
23        6        4   21         1
24        7        1   38         1
25        7        2   22         1
26        7        3   32         1
27        7        4   27         1
28        8        1   24         1
29        8        2   31         1
30        8        3   24         1
31        8        4   22         1
32        9        1   40         0
33        9        2   35         0
34        9        3   26         1
35        9        4   21         1
36       10        1   29         0
37       10        2   30         1
38       10        3   26         1
39       10        4   29         0

##QII d Compute the mean Body Mass Index (BMI), its standard error and 95% confidence interval.


In [39]:
#GIVEN PARAMETERS:

n = 40    # Total sample size
N = 9300  # Population size
A = 100   # Total number of clusters
B = 93    # Women per cluster
a = 10    # Clusters sampled
b = 4     # Women per cluster sampled

In [40]:
mean_BMI = populationTS['BMI'].mean()

In [44]:
# Step 1: Computing variance between clusters (S2)
cluster_means = populationTS.groupby('CLUSTER')['BMI'].mean()
S2 = np.var(cluster_means, ddof=1)  # Variance of cluster means

# Step 2: Computing variance within clusters
s2 = np.mean(populationTS.groupby('CLUSTER')['BMI'].var(ddof=1))  # Mean in-cluster variance

# Step 3: Computing the standard error formula for two-stage sampling
SE_mean_BMI = np.sqrt((1 / a) * S2 + (1 / (a * b)) * s2)


In [45]:
t_crit = t.ppf(0.975, df=a - 1)  # t critical value for 95% CI
CI_lower = mean_BMI - t_crit * SE_mean_BMI
CI_upper = mean_BMI + t_crit * SE_mean_BMI


In [47]:
print(mean_BMI)
print(SE_mean_BMI)
print(CI_lower, CI_upper)

29.0
1.3538012327434843
25.937488844268618 32.062511155731386


##Mean BMI : 29.0
##Stadard error : 1.35380
##Lower CI : 25.937488
##Upper CI: 32.062511

##QII e Estimate the standard error of the mean BMI if the sample consisted of a=5 clusters of b=4 each

In [48]:
a_new = 5
b_new = 4

In [49]:
SE_mean_BMI_new = np.sqrt((1 / a_new) * S2 + (1 / (a_new * b_new)) * s2)

In [51]:
print(SE_mean_BMI_new)

1.9145640641032504


##Answer : If the sample consisted of a=5 clusters of b=4 each, the standard error of the mean BMI would be 1.9145640

##QII g Now estimate the proportion of women who have ever been pregnant (EVERPREG=1) from the selected sample. Compute standard error, design effect deff and roh.

In [53]:
# Compute proportion of women who have ever been pregnant
p_hat = populationTS['EVERPREG'].mean()


np.float64(0.775)

In [55]:
# Computing between-cluster variance (S2 for EVERPREG)
cluster_means_everpreg = populationTS.groupby('CLUSTER')['EVERPREG'].mean()
S2_everpreg = np.var(cluster_means_everpreg, ddof=1)

In [56]:
# Compute within-cluster variance (s2 for EVERPREG)
s2_everpreg = np.mean(populationTS.groupby('CLUSTER')['EVERPREG'].var(ddof=1))

In [57]:
# Standard error
SE_p_hat = np.sqrt((1 / a) * S2_everpreg + (1 / (a * b)) * s2_everpreg)

In [58]:
#Design effect (DEFF)
deff = 1 + ((b - 1) * (S2_everpreg / (S2_everpreg + s2_everpreg)))

In [60]:
# Computing the intra-cluster correlation (roh, ρ)
roh = S2_everpreg / (S2_everpreg + s2_everpreg)

In [62]:
print(p_hat)
print(SE_p_hat)
print(deff)
print(roh)

0.775
0.09574271077563382
1.644859813084112
0.21495327102803738


## proportion of women who have ever been pregnant = 77.5%
## Standard error = 0.0957427
## Design effect = 1.64485981
##Intra-cluster correlation roh = 0.214953

## Q II h Compute the optimum cluster size if the cost per cluster is Ca = 100 dollars and the cost per observation within a cluster is Cb = dollars 10 for the proportion of women who have ever been pregnant (EVERPREG = 1)

In [64]:
c_a = 100  # Cost per cluster
c_b = 10   # Cost per observation within a cluster

# Compute the optimum cluster size
b_opt = np.sqrt((c_a * (1 - roh)) / (c_b * roh))

# Output the result
print(b_opt)

6.043321862224019


##Optimum cluster size is 6.043321

##Q 2 a = > 10:00
##b = 8.5/10